### PR TITLE
Build runtime (stable) with Lightning git tag in runtime/Makefile instead of latest_release in the plugin matrix

### PR DESCRIPTION
**Pull request by dime10**
Lightning tags have been explicitly specified in build recipes for a while, to ensure compatibility with a particular commit in the catalyst repo. Therefor it makes little sense to test configurations which are known not to work.

The plugin matrix will now build the runtime with a stable lightning config with respect to the lightning repo tag specified in our build scripts, which may be either `latest_release` or a commit in the history of the master branch.

### CODE REVIEW

**Review by maliasadi — APPROVED** [review]

File: .github/workflows/check-pl-compat.yaml [hunk 1]
```diff
@@ -116,21 +115,6 @@ jobs:
...
-    - name: Build Lightning Runtime (stable)
-      if: ${{ inputs.lightning == 'stable' }}
-      run: |
-        COMPILER_LAUNCHER="" \
-        C_COMPILER=$(which gcc }}) \
-        CXX_COMPILER=$(which g++ }}) \
-        RT_BUILD_DIR="$(pwd)/runtime-build" \
-        QIR_STDLIB_DIR="$(pwd)/qir-stdlib-build" \
-        QIR_STDLIB_INCLUDES_DIR="$(pwd)/qir-stdlib-build/include" \
-        LIGHTNING_GIT_TAG_VALUE=latest_release \
```
Comment (maliasadi) [inline .github/workflows/check-pl-compat.yaml:133]: Revert this to still build against stable Lightning

**Comment by maliasadi** [comment]
David, note that we are currently building the runtime with different lightning tags as the matrix is checking stable and latest versions of Lightning, Catalyst, and PennyLane. If you want to build lightning against one single tag then you should consider removing Lightning from this plugin matrix.

File: .github/workflows/check-pl-compat.yaml [hunk 2]
```diff
@@ -116,21 +115,6 @@ jobs:
         RT_BUILD_DIR="$(pwd)/runtime-build" \
         QIR_STDLIB_DIR="$(pwd)/qir-stdlib-build" \
         QIR_STDLIB_INCLUDES_DIR="$(pwd)/qir-stdlib-build/include" \
-        LIGHTNING_GIT_TAG_VALUE=master \
```
Comment (maliasadi) [inline .github/workflows/check-pl-compat.yaml:119]: As we discussed offline, let's keep both lightning tags (latest and stable) in the plugin matrix.

**Review by maliasadi — CHANGES_REQUESTED** [review]

File: .github/workflows/check-pl-compat.yaml (hunk 1 above)
Comment (dime10) [reply .github/workflows/check-pl-compat.yaml]: I'm closing this since it's outdated now :)

**Review by maliasadi — APPROVED** [review]
:raised_hands:

**Pull request merged**